pkg/crd/kubermatic/v1/preset.go

```diff
@@ -357,9 +380,10 @@ func (s Openstack) IsValid() bool {
 		return len(s.ApplicationCredentialSecret) > 0
 	}
 
+	projectOrTenant := (len(s.Project) > 0 || len(s.ProjectID) > 0) || (len(s.Tenant) > 0 || len(s.TenantID) > 0)
```

Comment:
s/projectOrTenant/hasProjectOrTenant

So the logic is any of these four is enough? Then you can remove the brackets as they do not change anything.

Reply:
done :)